### PR TITLE
React

### PR DESCRIPTION
**Pull request by MosabDwaikat**

### CODE REVIEW

File: src/components/AddTask/index.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        id="add-task-btn"
+        onClick={handleAddTask}
+      >
+        Add
+      </button>
+      <div className="add-task-info-panel">
+        <div className="add-task-info-header-panel">
+          <input
+            type="text"
+            className={"task-title" + (!title ? " input-error" : "")}
```
Comment (oalfroukh) [inline src/components/AddTask/index.tsx]: I prefer to enable this error only once try to add task and there is no title or description and not enable it always even though before submitting the task

File: src/components/List/index.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  setTasks: (tasks: TaskType[]) => void;
+  filter: PriorityType;
+}
+const List = ({ tasks, setTasks, filter }: ListProps) => {
+  const deleteTask = (id: string) => {
+    const newTasks = tasks.filter((task) => task.id !== id);
+    setTasks(newTasks);
+  };
+
+  const editTask = (updatedTask: (typeof tasks)[0]) => {
```
Comment (oalfroukh) [inline src/components/List/index.tsx]: what does `: (typeof tasks)[0]` mean?

File: src/components/Task/index.tsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+      case PriorityType.high:
+        return "bg-color-red";
+      case PriorityType.medium:
+        return "bg-color-yellow";
+      case PriorityType.low:
+        return "bg-color-green";
+      default:
+        return "";
+    }
+  };
```
Comment (oalfroukh) [inline src/components/Task/index.tsx]: there is another one on addTask!, Perhaps we can create a utils folder and put this function inside it?

File: src/components/List/index.tsx (hunk 2 above)
Comment (MosabDwaikat) [reply src/components/List/index.tsx]: i missed it, it was before i created the type

**Review by oalfroukh — APPROVED** [review]

**Pull request merged**